### PR TITLE
[BugFix] Shutdown update manager & tracer during BE and UT exit

### LINKED ISSUE #30828: ASAN memory leak in apply_thread_pool
### Real behavior (Required)

```
==44100==ERROR: LeakSanitizer: detected memory leaks

Direct leak of 72 byte(s) in 1 object(s) allocated from:
    #0 0xb0d59f7 in operator new[](unsigned long) ../../.././libsanitizer/asan/asan_new_delete.cpp:102
    #1 0xb6ffae3 in opentelemetry::v1::context::ThreadLocalContextStorage::Stack::Resize(unsigned long) (/root/StarRocks/be/ut_build_ASAN/test/starrocks_test+0xb6ffae3)
    #2 0xb6ff8f8 in opentelemetry::v1::context::ThreadLocalContextStorage::Stack::Push(opentelemetry::v1::context::Context const&) /var/local/thirdparty/installed/include/opentelemetry/context/runtime_context.h:271
    #3 0xb6ff245 in opentelemetry::v1::context::ThreadLocalContextStorage::Attach(opentelemetry::v1::context::Context const&) /var/local/thirdparty/installed/include/opentelemetry/context/runtime_context.h:214
    #4 0x112d984d in opentelemetry::v1::context::RuntimeContext::Attach(opentelemetry::v1::context::Context const&) /var/local/thirdparty/installed/include/opentelemetry/context/runtime_context.h:92
    #5 0x112d9ee3 in opentelemetry::v1::trace::Scope::Scope(opentelemetry::v1::nostd::shared_ptr<opentelemetry::v1::trace::Span> const&) /var/local/thirdparty/installed/include/opentelemetry/trace/scope.h:29
    #6 0x1b444655 in starrocks::TabletUpdates::_apply_normal_rowset_commit(starrocks::EditVersionInfo const&, std::shared_ptr<starrocks::Rowset> const&) /root/StarRocks/be/src/storage/tablet_updates.cpp:1067
    #7 0x1b442670 in starrocks::TabletUpdates::_apply_rowset_commit(starrocks::EditVersionInfo const&) /root/StarRocks/be/src/storage/tablet_updates.cpp:1043
    #8 0x1b43b8b0 in starrocks::TabletUpdates::do_apply() /root/StarRocks/be/src/storage/tablet_updates.cpp:847
    #9 0x1b4daad0 in starrocks::ApplyCommitTask::run() /root/StarRocks/be/src/storage/tablet_updates.cpp:786
    #10 0x1cf1fa8d in starrocks::ThreadPool::dispatch_thread() /root/StarRocks/be/src/util/threadpool.cpp:554
    #11 0x1cf4474a in void std::__invoke_impl<void, void (starrocks::ThreadPool::*&)(), starrocks::ThreadPool*&>(std::__invoke_memfun_deref, void (starrocks::ThreadPool::*&)(), starrocks::ThreadPool*&) /opt/rh/gcc-toolset-10/root/usr/include/c++/10.3.0/bits/invoke.h:73
    #12 0x1cf44114 in std::__invoke_result<void (starrocks::ThreadPool::*&)(), starrocks::ThreadPool*&>::type std::__invoke<void (starrocks::ThreadPool::*&)(), starrocks::ThreadPool*&>(void (starrocks::ThreadPool::*&)(), starrocks::ThreadPool*&) /opt/rh/gcc-toolset-10/root/usr/include/c++/10.3.0/bits/invoke.h:95
    #13 0x1cf42c7b in void std::_Bind<void (starrocks::ThreadPool::*(starrocks::ThreadPool*))()>::__call<void, , 0ul>(std::tuple<>&&, std::_Index_tuple<0ul>) /opt/rh/gcc-toolset-10/root/usr/include/c++/10.3.0/functional:416
    #14 0x1cf415b0 in void std::_Bind<void (starrocks::ThreadPool::*(starrocks::ThreadPool*))()>::operator()<, void>() /opt/rh/gcc-toolset-10/root/usr/include/c++/10.3.0/functional:499
    #15 0x1cf3e705 in void std::__invoke_impl<void, std::_Bind<void (starrocks::ThreadPool::*(starrocks::ThreadPool*))()>&>(std::__invoke_other, std::_Bind<void (starrocks::ThreadPool::*(starrocks::ThreadPool*))()>&) /opt/rh/gcc-toolset-10/root/usr/include/c++/10.3.0/bits/invoke.h:60
    #16 0x1cf39ee4 in std::enable_if<is_invocable_r_v<void, std::_Bind<void (starrocks::ThreadPool::*(starrocks::ThreadPool*))()>&>, void>::type std::__invoke_r<void, std::_Bind<void (starrocks::ThreadPool::*(starrocks::ThreadPool*))()>&>(std::_Bind<void (starrocks::ThreadPool::*(starrocks::ThreadPool*))()>&) /opt/rh/gcc-toolset-10/root/usr/include/c++/10.3.0/bits/invoke.h:110
    #17 0x1cf351fe in std::_Function_handler<void (), std::_Bind<void (starrocks::ThreadPool::*(starrocks::ThreadPool*))()> >::_M_invoke(std::_Any_data const&) /opt/rh/gcc-toolset-10/root/usr/include/c++/10.3.0/bits/std_function.h:291
    #18 0xb996924 in std::function<void ()>::operator()() const /opt/rh/gcc-toolset-10/root/usr/include/c++/10.3.0/bits/std_function.h:622
    #19 0x1cefff99 in starrocks::Thread::supervise_thread(void*) /root/StarRocks/be/src/util/thread.cpp:365
    #20 0x7f6a0a865ea4 in start_thread (/lib64/libpthread.so.0+0x7ea4)
```
### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`

### PR DESCRIPTION
**Pull request by decster**
Fixes #30828

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Review by sevev — DISMISSED** [review]

File: be/src/storage/update_manager.cpp [hunk 1]
```diff
@@ -119,6 +119,15 @@ Status UpdateManager::init() {
     return Status::OK();
 }
 
+void UpdateManager::stop() {
+    if (_get_pindex_thread_pool) {
```
Comment (trueeyu) [inline be/src/storage/update_manager.cpp:123]: already shutdown in destructor?

Comment (decster) [reply be/src/storage/update_manager.cpp:114]: ok shutdown in destructor removed

**Review by trueeyu — APPROVED** [review]

**Review by chaoyli — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Pull request merged**